design-proposals/sidecar-hooks.md

```diff
@@ -0,0 +1,56 @@
+# Sidecar hooks
+
+## Overview
+
+A hooking mechanism that allows customizations to the VMI before it starts without changing KubeVirt
+core components. The common user of those hooks are Sidecars that will run in the same Pod that the
+VMI will be deployed.
+
+## Motivation
+
+A way to accommodate use cases that we might not want to bring and support in KubeVirt itself or to
+give room to test and develop features while they are still not mature enough.
+
+## Goals
+
+- Provide and maintain a set of gRPC APIs for users to do modifications to the VMI
+- Provide and maintain sidecars that are important for core functionalities
+
+## Non Goals
+
+- Provide and maintain a wide variety of sidecars that are not part of KubeVirt's core
+  functionalities.
+
+## User Stories
+
+- As a VM owner, I want to apply custom configurations to QEMU command line
+  - For debugability, by adding support or tweak [to run with debug tools][]
+  - To test unsupported features or apply changes that [could workaround bugs][]
+- As a KubeVirt developer, I want to provide alternative solutions that might not be ready to core
```

Comment:
We want to provide a modular, plug-able solution, to Kubevirt without overloading the core components. 

It is not necessary a matter of readiness.

Reply:
Ah, do you disagree with it? I thought that `network binding plugins` was an example of this. Of course, I can be wrong.